app/src/main/java/com/isc/hermes/MainActivity.java

```diff
@@ -54,6 +64,49 @@ private void addMapboxSearcher() {
         searcherController.runSearcher();
     }
 
+    /**
+     *This function helps to give functionality to the side menu, so that it can be visible and hidden, when necessary.
+     *
+     * @param view Helps build the view.
+     */
+    public void openSideMenu(View view) {
+        LinearLayout lateralMenu = findViewById(R.id.lateralMenu);
+        if (!visibilityMenu) {
+            lateralMenu.setVisibility(View.VISIBLE);
+            visibilityMenu = true;
+            setMapScrollGesturesEnabled(false);
+        } else {
+            lateralMenu.setVisibility(View.GONE);
+            visibilityMenu = false;
+            setMapScrollGesturesEnabled(true);
+        }
+    }
+
+    /**
+     * Enables or disables map scroll gestures.
+     *
+     * @param enabled Boolean indicating whether to enable map scroll gestures.
+     */
+    private void setMapScrollGesturesEnabled(boolean enabled) {
+        mapView.getMapAsync(new OnMapReadyCallback() {
+            @Override
+            public void onMapReady(@NonNull MapboxMap mapboxMap) {
+                mapboxMap.getUiSettings().setScrollGesturesEnabled(enabled);
+            }
+        });
+    }
+
+    /**
+     * Logs out the current user and redirects to the login activity.
+     *
+     * @param view The view of the button that has been clicked.
+     */
+    public void logOut(View view){
+        googleAuthentication.signOut(this);
+        Intent intent = new Intent(MainActivity.this, SignUpActivityView.class);
```

Comment:
This functionality to return to the initial screen should be inside the LogOut method of the GoogleAuthentication class.

Reply:
Done